src/navigation/failure_identification/failure_identification.py

```diff
@@ -171,8 +174,14 @@ def update(self, nav_status: SmachContainerStatus, drive_status: MotorsStatus, o
         # publish the watchdog status if the nav state is not recovery
         if TEST_RECOVERY_STATE:
             self.stuck_publisher.publish(Bool(self.cur_stuck))
-        elif nav_status.active_states[0] != "recovery":
-            self.stuck_publisher.publish(Bool(self.watchdog.is_stuck(self._df)))
+        elif nav_status.active_states[0] != "RecoveryState":
+            if self.recovery_end_time is None or rospy.Time.now() - self.recovery_end_time > rospy.Duration(
+                POST_RECOVERY_GRACE_PERIOD
+            ):
+                self.stuck_publisher.publish(Bool(self.watchdog.is_stuck(self._df)))
+        else:
```

Comment:
Is this else supposed to go with the elif statement on 177 or the if statement on 178? I'm just confused as to why we would publish False and set recovery_end_time if nav_status.active_states[0] == "RecoveryState" (since to go in the else statement, the elif would have to be false, right?).

Reply:
I made the variable name more clear but basically the idea is that while we are recovering we should not report that we're stuck because we already know that we were stuck and we are currently backing out trying to get unstuck so that might involve effort, because I added the thing where we give up on recovering after 10 seconds (indicating that we are stuck in recovery) we will switch back to the previous state and in that time we want to give it a grace period to get unstuck so we don't immediately switch back to the recovery state right after we switched back. Hope that makes sense if it doesn't I can further clarify and we can talk about it

Reply:
Yes that makes much more sense! Now I see how it works. Thank you for the explanation.